crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -279,17 +279,31 @@ impl EventTimelineItem {
         }
     }
 
-    /// Flag indicating this timeline item can be edited by current user.
+    /// Flag indicating this timeline item can be edited by the current user.
     pub fn is_editable(&self) -> bool {
+        // This must be in sync with the early returns of `Timeline::edit`
+        //
+        if !self.is_own() {
+            // In theory could work, but it's hard to compute locally.
+            return false;
+        }
+
+        if self.event_id().is_none() {
+            // Local echoes without an event id (not sent yet) can't be edited.
```

Comment:
Why?

Reply:
Not technically infeasible (would require a new mechanism to do it in the send queue), but it's not been deemed necessary by the product team.

Reply:
OK, fair enough.